db/migration/1661264304751-MigrateSelectedData.ts

```diff
@@ -0,0 +1,118 @@
+        }
+
+        const newConfig = { ...legacyConfig } as LegacyGrapherInterface
+
+        // Migrate selected entities
+        newConfig.selectedEntityIds = uniq(
+            legacyConfig.selectedData.map((row) => row.entityId)
+        ) // We need to do uniq because an EntityName may appear multiple times in the old graphers, once for each dimension
+
+        // Migrate dimension and entity colors.
+        // Iterate through the reversed array because in case of multiple entries for one entity, the last one applies.
+        legacyConfig.selectedData
+            .slice()
+            .reverse()
+            .forEach((item) => {
+                if (item.entityId && item.color) {
+                    // migrate entity color
+                    if (!legacyConfig.selectedEntityColors) {
+                        newConfig.selectedEntityColors =
+                            newConfig.selectedEntityColors ?? {}
+                        const entityName = entityNameById[item.entityId]
+                        if (entityName) {
+                            newConfig.selectedEntityColors[entityName] ??=
+                                item.color
+                        }
+                    }
+
+                    // migrate dimension color
+                    const dimension = newConfig.dimensions?.[item.index]
+                    if (dimension?.variableId) {
+                        dimension.display = dimension.display ?? {}
+                        dimension.display.color ??= item.color
+                    }
+                }
+            })
+
+        const migrateDimensionsTypes: ChartTypeName[] = [
+            ChartTypeName.Marimekko,
+            ChartTypeName.StackedArea,
+            ChartTypeName.StackedBar,
+            ChartTypeName.StackedDiscreteBar,
+        ]
+
+        // Migrate order of dimensions.
+        // Only applied to stacked charts, because only they rely on `yColumnSlugsInSelectionOrder`.
+        if (
+            legacyConfig.type &&
+            migrateDimensionsTypes.includes(legacyConfig.type)
+        ) {
+            const variableIDsInSelectionOrder = excludeUndefined(
+                legacyConfig.selectedData?.map(
+                    (item) => legacyConfig.dimensions?.[item.index]?.variableId
+                ) ?? []
+            )
+
+            newConfig.dimensions = sortBy(newConfig.dimensions || [], (dim) =>
+                variableIDsInSelectionOrder.findIndex(
+                    (variableId) => dim.variableId === variableId
+                )
+            ).filter((dim) =>
+                variableIDsInSelectionOrder.includes(dim.variableId)
+            )
+        }
+
+        delete newConfig.selectedData
+
+        return newConfig
+    }
+
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        const tables = {
+            charts: "config",
+            chart_revisions: "config",
+            suggested_chart_revisions: "suggestedConfig",
+        }
```

Comment:
Just as a note, this migration runs on `charts`, `chart_revisions`, and `suggested_chart_revisions`.